docs/resources/application.md

```diff
@@ -37,9 +50,13 @@ output "application_key" {
 ### Required
 
 - `name` (String) Name of the Application
-- `permissions` (Set of String) Permissions for the Application
 - `type` (String) Type for the Application
 
+### Optional
```

Comment:
Is one of the below required?

Reply:
Either rules or permissions is required, but I don't think we can express that through the terraform provider. A 400 error will be returned if a user tries to create an app with neither, or with both.